scripts/devops_tasks/validate_formatting.py

```diff
@@ -18,63 +18,79 @@
 root_dir = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", ".."))
 sdk_dir = os.path.join(root_dir, "sdk")
 
+
 def run_black(service_dir):
     results = []
     logging.info("Running black for {}".format(service_dir))
 
-    discovered_packages = discover_targeted_packages("azure*", os.path.join(root_dir, "sdk", service_dir))
+    discovered_packages = discover_targeted_packages(
+        "azure*", os.path.join(root_dir, "sdk", service_dir)
+    )
 
     for package in discovered_packages:
         package_name = os.path.basename(package)
 
         if is_check_enabled(package, "black", True):
-            out = subprocess.Popen([sys.executable, "-m", "black", "--config", os.path.join(root_dir, "eng", "black-pyproject.toml"), os.path.join("sdk", service_dir, package_name)],
+            out = subprocess.Popen(
+                [
+                    "tox",
+                    "-qqq",
+                    "--conf",
+                    os.path.join(root_dir, "eng", "tox", "tox.ini"),
+                    "--root",
+                    root_dir,
```

Comment:
> I'm a little bit concerned that we will need to install black into each tox environment, but they'll share cache, so it shouldn't add much at all hopefully +1
> 
> I think it's worthwhile to keep things consistent.

I waffled as to whether it whether it was worthwhile to set `--root` to be package specific (i.e. the current implementation), or just set it to the repository root.

I chose to err on the side of caution (to avoid the potential for concurrency issues), but would change if asked.

Reply:
Nah, I'm good with it as-is. Please give this a hold until end of this week, then feel free to merge it.